src/main/scala/sigmastate/Values.scala

```diff
@@ -420,13 +420,17 @@ object Values {
     def deconstruct = immutable.Seq[Any](elementType) ++ items
 
     def reconstruct(cs: immutable.Seq[Any]) = cs match {
-      case Seq(t: SType, vs@_*) => ConcreteCollection[SType](vs.asInstanceOf[Seq[Value[V]]].toIndexedSeq)(t)
+      case Seq(t: SType, vs@_*) => new ConcreteCollection[SType](vs.asInstanceOf[Seq[Value[V]]].toIndexedSeq, t)
```

Comment:
What about removing Rewritable, it is not necessary any more

Reply:
Done.